### PR TITLE
Blog homepage

### PR DESCRIPTION
**Pull request by edwoodward**
Added API to retrieve [posts by subject

`self.client.get('/apps/cms/api/search/',{'subjects': 'Math'})`

Added interest_block to NewsIndex and removed unneeded fields

```
{
    "id": 90,
    "meta": {
        "slug": "openstax-news",
        "seo_title": "Get News and Updates Through the OpenStax Blog",
        "search_description": "Read original content from faculty, students, OER experts, and more about a variety of topics related to OER.",
        "type": "news.NewsIndex",
        "detail_url": "https://none.openstax.org/apps/cms/api/v2/pages/90/",
        "html_url": "http://openstax.org/openstax-news/",
        "show_in_menus": false,
        "first_published_at": "2016-06-24T07:33:22.796928-05:00",
        "parent": {
            "id": 29,
            "meta": {
                "type": "pages.HomePage",
                "detail_url": "https://none.openstax.org/apps/cms/api/v2/pages/29/",
                "html_url": "http://openstax.org/"
            },
            "title": "OpenStax Homepage"
        },
        "locale": "en"
    },
    "title": "Openstax News",
    "interest_block": [
        {
            "type": "heading",
            "value": "Interested in OpenStax?",
            "id": "ec046294-0520-425c-8eba-2d431c566c0e"
        },
        {
            "type": "description",
            "value": "We will show you how to take advantage of what OpenStax has to offer",
            "id": "73fcca51-f9fb-49c3-9cfc-e26c8e0aeab9"
        },
        {
            "type": "button_text",
            "value": "Sign up to learn more",
            "id": "810d6e8d-23d5-4d1d-b044-9be077433b01"
        },
        {
            "type": "button_href",
            "value": "https://openstax.org/login",
            "id": "7133d4b3-a500-4216-8897-35b277d4bf97"
        }
    ],
    "promote_image": null
}
```

### CODE REVIEW

File: news/migrations/0041_auto_20220517_1353.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('news', '0040_alter_newsarticle_content_types'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
```
Comment (mwvolo) [inline news/migrations/0041_auto_20220517_1353.py:15]: probably alright this time but remember that Dante said it's best to remove references in one PR, then do the migration in the next release

File: news/search.py [hunk 2]
```diff
@@ -75,7 +75,12 @@ def search(request):
         if ('subjects' in request.GET) and request.GET['subjects'].strip():
             subjects = request.GET['subjects'].split(',')
 
-        found_entries = news_article_search(collection_name, types, subjects)
+        found_entries = news_article_collection_search(collection_name, types, subjects)
+
+    elif ('subjects' in request.GET) and request.GET['subjects'].strip():
+        print('**Only subjects query')
```
Comment (mwvolo) [inline news/search.py:81]: remove print statement

**Review by mwvolo — APPROVED** [review]
approved with comments

File: news/search.py (hunk 2 above)
Comment (edwoodward) [reply news/search.py]: Removed

File: news/migrations/0041_auto_20220517_1353.py (hunk 1 above)
Comment (edwoodward) [reply news/migrations/0041_auto_20220517_1353.py:15]: I'll remember next time

**Pull request merged**